### PR TITLE
refactored ex NOOP filter & specs into filter base specs

### PR DESCRIPTION
**Pull request by colinsurprenant**
the NOOP filter is now a dynamically defined class into the `spec/filters/base_spec.rb` and the Plugin#lookup method has been adjusted to support dynamically defined plugins.

### CODE REVIEW

File: lib/logstash/plugin.rb [hunk 1]
```diff
@@ -124,32 +124,54 @@ def inspect
...
+    klass.ancestors.include?(LogStash::Plugin) && klass.respond_to?(:config_name) && klass.config_name == name
+  end
 
-    return klass
-  rescue LoadError => e
-    raise LogStash::PluginLoadingError,
-      I18n.t("logstash.pipeline.plugin-loading-error", :type => type, :name => name, :path => path, :error => e.to_s)
-  end # def load
+  # @return [Cabin::Channel] logger channel for class methods
+  def self.logger
```
Comment (jordansissel) [inline lib/logstash/plugin.rb]: Yeah the intent is to use LogStash::Logger for everything

**Comment by jordansissel** [comment]
@colinsurprenant and I chatted today about Logger stuff and agreed on some unrelated improvements to the internal logger stuff (will file tickets about this).

+1 LGTM this change.

**Comment by colinsurprenant** [comment]
ok to test

**Comment by ph** [comment]
test this please

**Pull request closed without merging**